packages/asset-swapper/src/utils/market_operation_utils/sampler_operations.ts

```diff
@@ -327,41 +368,71 @@ export const samplerOperations = {
         takerToken: string,
         makerFillAmounts: BigNumber[],
         liquidityProviderRegistryAddress?: string | undefined,
+        fakeBuyOpts: FakeBuyOpts = DEFAULT_FAKE_BUY_OPTS,
     ): BatchedOperation<DexSample[][]> {
-        const subOps = sources.map(source => {
-            if (source === ERC20BridgeSource.Eth2Dai) {
-                return samplerOperations.getEth2DaiBuyQuotes(makerToken, takerToken, makerFillAmounts);
-            } else if (source === ERC20BridgeSource.Uniswap) {
-                return samplerOperations.getUniswapBuyQuotes(makerToken, takerToken, makerFillAmounts);
-            } else if (source === ERC20BridgeSource.LiquidityProvider) {
-                if (liquidityProviderRegistryAddress === undefined) {
-                    throw new Error(
-                        'Cannot sample liquidity from a LiquidityProvider liquidity pool, if a registry is not provided.',
+        const subOps = sources
+            .map(source => {
+                let batchedOperation;
+                if (source === ERC20BridgeSource.Eth2Dai) {
+                    batchedOperation = samplerOperations.getEth2DaiBuyQuotes(makerToken, takerToken, makerFillAmounts);
+                } else if (source === ERC20BridgeSource.Uniswap) {
+                    batchedOperation = samplerOperations.getUniswapBuyQuotes(makerToken, takerToken, makerFillAmounts);
+                } else if (source === ERC20BridgeSource.Kyber) {
+                    batchedOperation = samplerOperations.getKyberBuyQuotes(
+                        makerToken,
+                        takerToken,
+                        makerFillAmounts,
+                        fakeBuyOpts,
+                    );
+                } else if (Object.keys(DEFAULT_CURVE_OPTS).includes(source)) {
```

Comment:
I feel like I've seen this check before. Should we turn this into a function (`isCurveSource()`) so it's more clear what's going on here?